test/MinImports.lean

```diff
@@ -69,3 +75,59 @@ import Mathlib.Data.Nat.Notation
 #guard_msgs in
 #min_imports in
 lemma hi (n : ℕ) : n = n := by extract_goal; rfl
+
+/--
+warning: Imports increased to
+[Init.Guard, Lean.Parser.Term, Mathlib.Data.Int.Notation]
+note: this linter can be disabled with `set_option linter.minImports false`
+-/
+#guard_msgs in
+set_option linter.minImports true in
+#guard (0 : ℤ) = 0
+
+#guard_msgs in
+-- no new imports needed here, so no message
+set_option linter.minImports true in
+#guard (0 : ℤ) = 0
+
+set_option linter.minImports false in
+#reset_min_imports
+
+/--
+warning: Imports increased to
+[Init.Guard, Lean.Parser.Term, Mathlib.Data.Int.Notation]
+note: this linter can be disabled with `set_option linter.minImports false`
+-/
+#guard_msgs in
+-- again, the imports pick-up, after the reset
+set_option linter.minImports true in
+#guard (0 : ℤ) = 0
+
+-- finally, we leave the linter "on", to give it a chance to parse `eoi`
+-- and to silence it, we make sure to put in a command for each import
+
+/--
+warning: Imports increased to
+[Init.Guard, Lean.Parser.Command, Mathlib.Data.Int.Notation]
+note: this linter can be disabled with `set_option linter.minImports false`
+-/
+#guard_msgs in
+set_option linter.minImports true
```

Comment:
Will `set_option linter.minImports true` always cause the next command to get a warning?
Should there be a command that sets `minImportsRef` to whatever is currently needed, without getting a warning from *this* linter. (It will be caught during PR review by the linter that flags commands.)

Reply:
This is a little tricky: the linter knows what are the imports-so-far, only if it has been diligently increasing its "import counter" with every command that it saw.  A switch like the one that you are suggesting would mean that the linter should have updated its counter also when it was "off".

I think that this feature is not for this linter in its current form.  If you want to know about whether a specific command increases imports, then you activate the linter at the start of the file and see if this command gets flagged.  If you want to know the imports of just the current command, then you can use `#min_imports in`.  Does that make sense?

Reply:
What I have not yet decided, is whether the linter should consider `set_option`s as something that increases imports.  In some sense, it currently does and does not: it notices that a `set_option` is a command and adds `Lean.Parser.Command` to the imports, but it does not scan the state for what option it is, so it does not find the file where the option is defined.

Reply:
Re-reading your comment, it could also be interpreted as: can we make sure that the first command after `set_option linter.minImports true` is *never* flagged and considered like a "reference" command.  This could be done: for instance the `minImportsRef` could be an `Option NameSet`, starting with `none` and, when it changes from `none` to `some ...`, the linter says nothing, but a change from `some ...` to `some (something else)` makes the linter speak.